docs/source/architecture/data_services_architecture.rst

```diff
@@ -1,4 +1,110 @@
+
+#. Metadata Builder: It is responsible for generating JDO metadata for entity.
+
+#. Enum Builder: Value defined in Combo Box/ Field Metadata will be used to create an Enum.
+
+#. Lookup Builder: It allows to create a method without body and added into javassist service interface and method with body added into javassist service class.
+
+#. Processors: Processors process DDEs based on annotations and create Infrastructure and Metadata Information for those entities which can used via DataNucleus enhancer to enhance entities to use them for Persistence Operations.
+
+#. JDO Enhancer: It extends DataNuclues JDOEnhancer. It enhances classes to add missing information so that the classes can be used by DataNuclues for persistance related operations.
+
+#. MDS Data Provider: It creates Mds data provider json to register in task module.
+
+#. Infrastructure Classes, EUDE and DDE: It represents all entities, services, repositories of entities used by MDS Web module to represent MDS entities in UI.
+
+#. Pre Defined Service Section:
+Mds provide some predefined services used by Mds Web and other modules of motech platform :
+
+    *Action Handler Service: It provides methods for handling tasks actions events related
+    with MDS CRUD operations .i.e. create, update and delete.
+
+    * Bundle watcher Suspension Service: An OSGi service, allowing to temporarily disable bundle processing by MDS. When processing gets suspended, MDS will still listen to bundle events, but instead of processing them, they will be queued and processed after the processing gets restored. This allows, besides others, to install/uninstall a larger amount of bundles at one time, without facing annotation processing problems.
+
+    * Csv Export Customizer: It provides custom methods to format related instances during csv export.
+
+    * Csv Import Customizer: It provides custom methods for searching, creating and updating instances during csv import.
+
+    * Csv Import Export Customizer: Service providing methods for importing and exporting instances using csv.
+
+    * Motech Data Service: Centre of Data Access Layer is Generic Data Access Pattern Class which defines generic persistence related methods. Superclass for defining service of any Entity in MDS.
+
+    * Entity Service: It defines CRUD actions on entities.
+
+    * History Service: It persists changes made on given instance. It captures instances which are not in trash.
+
+    * Trash Service: It provides methods related to trash mode (By Default, trash mode is active and can be turned off by user.)
+
+    * Jar Generator Service: It creates a bundle jar with all entities defined in MDS.
+
+    * Json LookUp Service: Service for managing lookups from JSON.
+
+    * MDS Bundle Regeneration Service: It provides methods to regenerate bundle and regenerate bundle after dde enhancement. This service use Jar Generator Service to regenerate bundle.
```

Comment:
On the topic of this service, we should describe how we use an event topic to make sure all clustered MDS instances regenerate their schema - which is the only real purpose of this service.